src/components/Personalization/createComponent.js

```diff
@@ -39,25 +39,34 @@ export default ({
           return;
         }
 
-        const personalization = createPersonalizationDetails({
+        const personalizationDetails = createPersonalizationDetails({
           renderDecisions,
           decisionScopes,
           event,
           viewCache
         });
 
-        if (personalization.shouldFetchData()) {
-          pageLoadHandler({
-            personalization,
+        if (personalizationDetails.shouldFetchData()) {
+          const decisionsDeferred = defer();
+
+          viewCache.storeViews(decisionsDeferred.promise);
+
+          fetchDataHandler({
+            decisionsDeferred,
```

Comment:
It would probably be simpler to just inject `viewCache` directly into `createResponseHandler` from index.js and had `createResponseHandler` call `viewCache.storeViews` with the decisions rather than a promise.